### ISSUE TITLE
nested heredocs

### ISSUE THREAD

**Issue by schmurfy** [atom/language-ruby#109]
That is a unusual sight but I needed to do it in a project and found out it is highlighted incorrectly, I looked at the grammar but I am not sure how to fix it if it can even be done :(

```ruby
def some_function(str1, str2)
  p [str1, str2]
end


some_function(<<-EOS1, <<-EOS2)
string 1
EOS1
string 2
EOS2
```

This code is valid, you can try executing it but the part between EOS1 and EOS2 has an incorrect highlighting.
I think the most sensible thing in this case would be to consider everything between EOS2 and EOS2 as string, is there any grammar guru around here ?:p

PS: github also get the highlighting wrong ^^

**Comment by mamenama** [comment]
Edited the ruby.cson file to add this in 5789cbffb5d5a6ce4355f5c1e5112f683bcd5a4a

In that version, all the heredoc is highlighted until the last heredoc (above, EOS2) is reached.

The closing parenthesis, alas, remains considered part of the string. There's probably some way around that, but it's probably more complicated.

**Comment by 50Wliu** [comment]
Fixed by #113.

**Comment by schmurfy** [comment]
thanks ! :)

The parenthesis considered inside the string is really a minor issue compared to how the code was shown as before ^^
